File: 2529-maximum-count-of-positive-integer-and-negative-integer/2529-maximum-count-of-positive-integer-and-negative-integer.cpp
class Solution {
public:
    int maximumCount(vector<int>& nums) {
        int ne=0;
        int po=0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]==0) continue;
            if(nums[i]>0) po++;
            else ne++;
        }
        if(po==ne) return po;
        else if(po>ne) return po;
        else return ne;
    }
};

File: 2529-maximum-count-of-positive-integer-and-negative-integer/2529-maximum-count-of-positive-integer-and-negative-integer.java
class Solution {
    public int maximumCount(int[] nums) {
        int n = nums.length;
        // finding last negative index
        int lo = 0;
        int hi = n-1;
        int nidx = -1;
        while(lo<=hi){
            int mid = lo + (hi-lo)/2;
            if(nums[mid]<0){
                nidx = mid;
                lo = mid+1;
            }
            else hi = mid-1;
        }
        //Now, finding first positive idx
        lo=0;
        hi=n-1;
        int pidx = -1;
        while(lo<=hi){
            int mid = lo + (hi-lo)/2;
            if(nums[mid]>0){
                pidx = mid;
                hi = mid-1;
            }
            else lo = mid+1;
        }
        if(pidx == -1) pidx = 0;
        else  pidx = n-pidx;
        if(nidx == -1) nidx = 0;
        else nidx = nidx+1;
        return Math.max(pidx,nidx);
    }
}